### PR TITLE
fix: support downgrading dependencies based on versions.json

### PR DESCRIPTION
**Pull request by platosha**
For dependencies without custom versions in `package.json` (when `dependencies` and `vaadin.dependencies` agree), currently a pinned version from `versions.json` is not applied if it is for some reason older than the one specified in `package.json`. This effectively prevents downgrading npm package versions by changing the platform dependency version.

This change removes the downgrade prevention check.

Context: to speed up the upcoming release, Hilla platform 2.2 reverted to using stable React and Vaadin components 24.1.x before the beta1 release after using 24.2.0.alphaX during alpha releases. At the moment upgrading a Hilla application from 2.2.0.alpha10 to beta1 does not downgrade the component dependencies back to 24.1, while this is expected.

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/VersionsJsonFilter.java [hunk 1]
```diff
@@ -82,9 +82,6 @@ JsonObject getFilteredVersions(JsonObject versions, String versionOrigin) {
                             versions.getString(key));
                 }
                 json.put(key, userManagedDependencies.getString(key));
-            } else if (vaadinDepsVersion != null
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/VersionsJsonFilter.java:85]: Code block 
```
            final FrontendVersion vaadinDepsVersion = FrontendUtils
                    .getPackageVersionFromJson(vaadinVersions, key,
                            "/package.json -> { vaadin { dependencies }}");
```
is not needed now.

**Review by mshabarov — CHANGES_REQUESTED** [review]
This change looks fine to me.

If I understand the algorithm properly, the previous approach used the platform pinned dependency if it is newer and closes the eyes on the fact that it comes from old platform.

This in my opinion is not correct. This adds a mess into dependencies. And maybe that's at least particularly why vaadin dance is needed.

Whenever you change platform version, you expect that your project will use all the versions provided by this concrete platform version, unless you override it by yourself.

I can't imagine at the moment in what circumstances this patch could break existing applications. Anyway, vaadin dance to the rescue while we are figuring out the use case where this patch breaks an app.

**Comment by mshabarov** [comment]
`runPnpmInstall_userVersionNewerThanPinned_installedOverlayVersionIsNotSpecifiedByPlatform` exactly checks that the dependency version in package.json is newer than pinned platform version and thus not touched. The expectation should be changes to opposite.

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/VersionsJsonFilter.java (hunk 1 above)
Comment (platosha) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/VersionsJsonFilter.java:85]: removed

**Review by mshabarov — APPROVED** [review]

**Pull request merged**